go-controller/pkg/node/gateway_localnet_linux_test.go

```diff
@@ -394,18 +389,20 @@ var _ = Describe("Node Operations", func() {
 				err = nodenft.MatchNFTRules(expectedNFT, nft.Dump())
 				Expect(err).NotTo(HaveOccurred())
 
-				fakeOvnNode.start(ctx,
-					&v1.ServiceList{
-						Items: []v1.Service{
-							service,
-						},
-					},
-				)
-
-				fNPW.watchFactory = fakeOvnNode.watcher
-				Expect(startNodePortWatcher(fNPW, fakeOvnNode.fakeClient, &fakeMgmtPortConfig)).To(Succeed())
+				stopChan := make(chan struct{})
+				fakeClient := util.GetOVNClientset(&service).GetNodeClientset()
+				wf, err := factory.NewNodeWatchFactory(fakeClient, "node")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(wf.Start()).To(Succeed())
+				defer func() {
+					close(stopChan)
+					wf.Shutdown()
```

Comment:
this is kind of boilerplate code in every test, but seems like we can't avoid as each test dealing with different service and endpointslice objs. hope that is ok.

Reply:
It certainly could be better. But I have to contain the changes to reasonable fixes to the lint issues, otherwise the work would be unbearable, and the review too!